includes/route.php

```diff
@@ -385,4 +385,13 @@ public function my_events() {
 	public static function convertToTimezone( string $date_time, string $time_zone ): string {
 		return ( new DateTime( $date_time, new DateTimeZone( 'UTC' ) ) )->setTimezone( new DateTimeZone( $time_zone ) )->format( 'Y-m-d H:i:s' );
 	}
+
+	/**
+	 * Get the base breadcrumb for events page.
+	 *
+	 * @return array The base breadcrumb.
+	 */
+	public static function get_base_breadcrumb(): array {
```

Comment:
Instead of adding a `get_base_breadcrumb()` function we could add a `set_breadcrumb()` function which would contain all the logic for breadcrumbs. Something like the following:

```php
public static function set_breadcrumb( string $name = '' ) {
	if ( '' === $name ) {
		gp_breadcrumb( array( __( 'Events' ) ) );
		return;
	}

	$breadcrumb   = array( gp_link_get( gp_url( '/events' ), __( 'Events' ) ) );
	$breadcrumb[] = esc_html( $name );
	gp_breadcrumb( $breadcrumb );
}
```

Then the templates can simply call:

```php
Route::set_breadcrumb( $event_title );
```

I think this would make for more maintainable code because the logic is concentrated on a single location instead of being in all templates.